src/Api/HL/RSQLInput.php

```diff
@@ -78,186 +78,266 @@ public function getOperators(): array
-                    [
-                        'OR' => [
-                            [$DB::quoteName($a) => ''],
-                            [$DB::quoteName($a) => null],
+                [
+                    // Empty string or null
+                    'operator' => '=empty=',
+                    'description' => 'is empty',
+                    'sql_where_callable' => fn ($a, $b) => [
+                        [
+                            'OR' => [
+                                [$DB::quoteName($a) => ''],
+                                [$DB::quoteName($a) => null],
+                            ]
                         ]
-                    ]
+                    ],
                 ],
-            ],
-            [
-                'operator' => '=notempty=',
-                'description' => 'is not empty',
-                'sql_where_callable' => fn ($a, $b) => [
-                    [
-                        'AND' => [
-                            [$DB::quoteName($a) => ['<>', '']],
-                            [$DB::quoteName($a) => ['<>', null]],
+                [
+                    'operator' => '=notempty=',
+                    'description' => 'is not empty',
+                    'sql_where_callable' => fn ($a, $b) => [
+                        [
+                            'AND' => [
+                                [$DB::quoteName($a) => ['<>', '']],
+                                [$DB::quoteName($a) => ['<>', null]],
+                            ]
                         ]
-                    ]
+                    ],
                 ],
-            ],
-        ];
+            ];
+        }
+        return $operators;
     }
 
-    /**
-     * @return array
-     * @phpstan-return {field: string, value: string, callable: callable}[]
-     */
-    private function getFilters()
+    public function getGroupedFilters(string $filter_string, array $flattened_props)
     {
-        $filters = [];
-        $rsql_filters = explode(';', $this->filter_string);
-        $operators = $this->getOperators();
-        $operator_pattern = '(' . implode('|', array_column($operators, 'operator')) . ')';
-        foreach ($rsql_filters as $rsql_filter) {
-            if (empty($rsql_filter)) {
+        // RSQL filters may be grouped with parenthesis
+        // Filters separated by a comma are OR'd together
+        // Filters separated by a semicolon are AND'd together
+        // When there is a mix of commas and semicolons within a group, the semicolons take precedence
+
+        // Parse the RSQL filter string into an array similar to the DBIterator criteria syntax
+        // For example, the RSQL filter string: "name==test1;name==test2,name==test3" would be parsed into:
+        // [0 => ['name==test1'], 'OR' => ['name==test2', 'name==test3']]
+        // The string: ("name==test1;name==test2),name==test3" would be parsed into:
+        // [0 => ['name==test1', 'name==test2'], 'OR' => ['name==test3']]
```

Comment:
`(name==test1 AND nam==test2) OR name==test3` have to be passed this way to `DBmysql::request()`:
```php
$where = [
    'OR' => [
        [
            ['name' => 'test1'],
            ['name' => 'test2'],
        ],
        ['name' => 'test3']
    ]
];
```

```suggestion
        // The string: "(name==test1;name==test2),name==test3" would be parsed into:
        // ['OR' => [['name==test1', 'name==test2'], ['name==test3']]]
```